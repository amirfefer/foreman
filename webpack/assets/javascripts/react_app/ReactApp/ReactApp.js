import React from 'react';
import PropTypes from 'prop-types';
import { Router } from 'react-router-dom';
import history from '../history';

import Layout from '../components/Layout';
import AppSwitcher from '../routes';

const ReactApp = ({ data }) => (
  <Router history={history}>
    <AppSwitcher data={data} />
  </Router>
);

ReactApp.propTypes = {
  data: PropTypes.shape({
    layout: Layout.propTypes.data,
    metadata: PropTypes.object,
  }).isRequired,
};

export default ReactApp;
