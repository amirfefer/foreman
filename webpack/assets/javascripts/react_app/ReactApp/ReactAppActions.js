import {
  APP_FETCH_SERVER_PROPS,
  UPDATE_LEGACY_LOADING_STATE,
} from './ReactAppConstants';

export const initializeMetadata = metadata => ({
  type: APP_FETCH_SERVER_PROPS,
  payload: metadata,
});

export const updateLegacyLoading = state => ({
  type: UPDATE_LEGACY_LOADING_STATE,
  payload: state,
});
