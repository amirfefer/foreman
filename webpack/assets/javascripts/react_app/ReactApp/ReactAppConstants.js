export const APP_FETCH_SERVER_PROPS = 'APP_FETCH_SERVER_PROPS';
export const UPDATE_LEGACY_LOADING_STATE = 'UPDATE_LEGACY_LOADING_STATE';
