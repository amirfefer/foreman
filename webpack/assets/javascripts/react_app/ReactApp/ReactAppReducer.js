import Immutable from 'seamless-immutable';
import { LOCATION_CHANGE } from 'connected-react-router';
import {
  APP_FETCH_SERVER_PROPS,
  UPDATE_LEGACY_LOADING_STATE,
} from './ReactAppConstants';

const initialState = Immutable({
  metadata: {},
  legacyLoading: false,
  // Private state's record, for current location please use `state.router.location`
  _currentHref: null,
  referer: null,
});

export default (state = initialState, action) => {
  const { payload, type } = action;

  switch (type) {
    case LOCATION_CHANGE:
      if (state._currentHref === window.location.href) return state;
      return state.merge({
        referer: state._currentHref,
        _currentHref: window.location.href,
        legacyLoading: true,
      });
    case UPDATE_LEGACY_LOADING_STATE:
      return state.set('legacyLoading', payload);
    case APP_FETCH_SERVER_PROPS:
      return state.set('metadata', payload);

    default:
      return state;
  }
};
