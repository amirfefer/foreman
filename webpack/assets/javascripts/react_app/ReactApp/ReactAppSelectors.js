export const selectReactApp = state => state.app;
export const selectReactAppMetadata = state => selectReactApp(state).metadata;
export const selectI18NReady = state => selectReactApp(state).i18nReady;

export const selectReactAppVersion = state =>
  selectReactAppMetadata(state).version;
export const selectReactAppPerPageOptions = state =>
  selectReactAppMetadata(state).perPageOptions;
