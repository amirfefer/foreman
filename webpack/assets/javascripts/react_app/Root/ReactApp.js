import React from 'react';
import PropTypes from 'prop-types';
import { ConnectedRouter } from 'connected-react-router';
import history from '../history';
import { getForemanContext } from '../Root/Context/ForemanContext';
import Layout, { propTypes as LayoutPropTypes } from '../components/Layout';
import AppSwitcher from '../routes';

const ReactApp = ({ data: { layout, metadata } }) => {
  const ForemanContext = getForemanContext(metadata);
  return (
    <ForemanContext.Provider value={metadata}>
      <ConnectedRouter history={history}>
        <Layout data={layout}>
          <AppSwitcher />
        </Layout>
      </ConnectedRouter>
    </ForemanContext.Provider>
  );
};

ReactApp.propTypes = {
  data: PropTypes.shape({
    layout: LayoutPropTypes.data,
    metadata: PropTypes.object.isRequired,
  }).isRequired,
};

export default ReactApp;
