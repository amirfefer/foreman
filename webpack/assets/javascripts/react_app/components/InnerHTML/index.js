import React, { useEffect, useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import DelayedSkeleton from '../DelayedSkeleton';

const DangerouslyInnerHTML = ({ html }) => {
  const divRef = useRef(null);
  const dispatch = useDispatch();
  const isLoading = useSelector(state => state.app.legacyLoading);
  useEffect(() => {
    const newContent = () => {
      const range = document.createRange();
      const parsedHtml = range.createContextualFragment(html);
      return parsedHtml;
    };
    if (html) {
      const oldContent = document.getElementById('content');
      const parsedHtml = newContent();
      if (oldContent) {
        divRef.current.replaceChild(parsedHtml, oldContent);
      } else divRef.current.appendChild(parsedHtml);
    }
  }, [html, dispatch]);
  return (
    <React.Fragment>
      <div
        id="main"
        style={{ display: isLoading ? 'none' : 'inherit' }}
        ref={divRef}
      />
      {isLoading && <DelayedSkeleton count={5} />}
    </React.Fragment>
  );
};

export default DangerouslyInnerHTML;

DangerouslyInnerHTML.propTypes = {
  html: PropTypes.string,
};

DangerouslyInnerHTML.defaultProps = {
  html: '',
};
