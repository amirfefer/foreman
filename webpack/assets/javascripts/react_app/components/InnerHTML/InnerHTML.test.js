import React from 'react';
import { IntegrationTestHelper } from '@theforeman/test';
import InnerHTML from './';
import { reducers } from '../../ReactApp';

const content =
  "<script>tfm.test() </script><div id='test'> <span> Hello World </span> </div> ";

describe('AuditsPage', () => {
  it('rendering', () => {
    const integrationTestHelper = new IntegrationTestHelper(reducers);

    const component = integrationTestHelper.mount(<InnerHTML html={content} />);
    expect(component).toMatchSnapshot();
  });
});
