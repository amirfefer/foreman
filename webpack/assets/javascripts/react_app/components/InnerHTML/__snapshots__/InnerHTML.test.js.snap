// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditsPage rendering 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <DangerouslyInnerHTML
    html="<script>tfm.test() </script><div id='test'> <span> Hello World </span> </div> "
  >
    <div
      id="main"
      style={
        Object {
          "display": "inherit",
        }
      }
    />
  </DangerouslyInnerHTML>
</Provider>
`;
